vtty.c

```diff
@@ -671,33 +675,41 @@ static struct file_operations vtmx_fops = {
 
 };
 
-static struct miscdevice vtmx_miscdev = {
-	.minor = MISC_DYNAMIC_MINOR,
-	.name = "vtmx",
-	.fops = &vtmx_fops
-};
-
-
+static struct miscdevice* vtmx_miscdev = NULL;
```

Comment:
This is a residue of my try, to make things more dynamically allocatable, but I got stuck at the VTTY_MAX value. 

In my IoT setup, I'll need more than 16 virtual terminals (120+). Currently I see a couple of ways to go: 
   * give the module at load time the maximum number of slave ports -> need dynamic allocation in `vtty_init()`
   * load multiple instances of the same module with different names for the master device -> the shown approach
   * always compile a custom version of the module for my setup, instead of a canonical solution

So in the first turn I went to assign a different name instead of `vtmx` at module load time and prepared the dynamic allocation stuff by turning the structure into a pointer to structure, going later to replace the static local variable by an allocation. Got stuck at VTTY_MAX again and was somewhat blind of the steps, that I already took, thus this residue. 

So, yes, there was a reason for turning this into a pointer: I went for dynamic allocation of structures and I want to assign the master device name in `vtty_init()`.

Reply:
Wow, that's a lot of virtual ports. I didn't expect applications like this. I would probably check what the pty driver does and try to follow their implementation. 

But that's an entirely different topic, so please remove this "residue" - as you called it - from this branch.

Reply:
I had a lot of bad experiences using a `cuse` approach. Then your little module came in handy and was much more stable from the start on.

I can change it back.